turbomoleio/core/molecule.py

```diff
@@ -511,6 +604,419 @@ def from_dict(cls, d):
+            add_user_def_bonds (bool): if True the pair of atoms will be added to the used defined bonds
+        """
+        if np.ndim(atom1) == 0:
+            atom1 = [atom1]
+        if np.ndim(atom2) == 0:
+            atom2 = [atom2]
+        if np.ndim(atom3) == 0:
+            atom3 = [atom3]
+        if np.ndim(atom4) == 0:
+            atom4 = [atom4]
+
+        if not len(atom1) == len(atom2) == len(atom3) == len(atom4):
+            raise ValueError("list of atoms should have the same length")
+
+        self._check_index(atom1)
+        self._check_index(atom2)
+        self._check_index(atom3)
+        self._check_index(atom4)
+
+        indices = np.transpose([atom1, atom2, atom3, atom4]).tolist()
+
+        self.int_def.append(DihedralAngle(status=status, indices=indices, weights=weights, value=value))
+
+        if add_user_def_bonds:
+            for (i, j, k, l) in indices:
+                self.user_defined_bonds.add((i, "-", j))
+                self.user_defined_bonds.add((j, "-", k))
+                self.user_defined_bonds.add((k, "-", l))
+
+    def get_int_def_inconsistencies(self, atol=1e-4, ltol=1e-4):
+        """
+        Gives a list of the internal coordinates whose values are not consistent
+        with the values obtained from the molecule object.
+
+        Args:
+            atol (float): tolerance allowed on distances (in bohr).
+            ltol (float): tolerance allowed on angles (in degrees).
+
+        Returns:
+            list: list of inconsistent internal coordinates.
+        """
+        errors = []
+
+        for idef in self.int_def:
+            if idef.coord_type == "length":
+                tol = ltol
+            elif idef.coord_type == "angle":
+                tol = atol
+            else:
+                raise ValueError("unknow coord_type {}".format(idef.coord_type))
+            if not idef.is_valid(self.molecule, tol):
+                errors.append(idef)
+
+        return errors
+
+    def has_inconsistencies(self, atol=1e-4, ltol=1e-4):
+        """
+        Checks if any inconsistency in internal coordinates is present.
+
+        Args:
+            atol (float): tolerance allowed on distances (in bohr).
+            ltol (float): tolerance allowed on angles (in degrees).
+
+        Returns:
+            bool: True if inconsistencies are present.
+        """
+        return len(self.get_int_def_inconsistencies(atol=atol, ltol=ltol)) != 0
+
+    def as_dict(self):
+        """
+        A JSON serializable dict representation of an object.
+        Overwrites base method to handle sets.
+
+        Returns:
+            dict: json representation of the object.
+        """
+        # sort sets to keep consistent representations of dictionary
+        d = {"@module": self.__class__.__module__, "@class": self.__class__.__name__,
+             "molecule": self.molecule.as_dict(), "int_def": [i.as_dict() for i in self.int_def],
```

Comment:
replace `molecule` with `structure` in the dict methods.

Reply:
Done, I've also removed int_def and user_defined_bonds from them.